pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -426,6 +426,16 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			}).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, initializeShootClients, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "DestroyNodeLocalDNS system component",
+			Fn:           flow.TaskFn(botanist.DestroyNodeLocalDNS).SkipIf(o.Shoot.HibernationEnabled || o.Shoot.NodeLocalDNSEnabled),
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, initializeShootClients, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying NodeLocalDNS system component",
```

Comment:
```suggestion
			Name:         "Deploying node-local-dns system component",
```

Reply:
CoreDNS have followed similar pattern, so I think this should not be an issue.

Reply:
I think that one was missed. But all others follow this pattern.